src/main/kotlin/org/web3j/quorum/enclave/protocol/ipc/EnclaveIpcService.kt

```diff
@@ -2,62 +2,56 @@ package org.web3j.quorum.enclave.protocol.ipc
 
 import com.fasterxml.jackson.annotation.JsonInclude
 import com.fasterxml.jackson.module.kotlin.*
-import org.web3j.protocol.ipc.IOFacade
+import okhttp3.MediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import org.web3j.quorum.enclave.EnclaveClientConnectionException
 import org.web3j.quorum.enclave.protocol.EnclaveService
-import org.web3j.quorum.enclave.protocol.utils.RequestBuilder
-import org.web3j.quorum.enclave.protocol.utils.ResponseParser
 
 /**
  * IPC service layer
  */
-abstract class EnclaveIpcService : EnclaveService {
+class EnclaveIpcService(private val url: String, private val port: Int, private val client: OkHttpClient = OkHttpClient()) : EnclaveService {
```

Comment:
What is the purpose of the url and port for the IpcService?
As far as I can see it is the user's concern to initialize an OkHttpClient with a UnixDomainSocketFactory. Was that the intention? If it was - I suspect the EnclaveHttpService could also be used with a user initialized OkHttpClient.

Reply:
As far as I know okHttp only work using an uri. So when we want to send a new request to Constellation we need `url` and `port`.